bpf/lib/nodeport.h

```diff
@@ -646,6 +631,23 @@ static inline int nodeport_lb4(struct __sk_buff *skb, __u32 src_identity)
 		return TC_ACT_OK;
 	}
 
+	if (!svc->is_k8s_external_ip) {
```

Comment:
How does this set handle the case where you SNAT the external request to the remote backend and at the same time an app from the node itself directly connects to the same backend where kernel picks same source port. In other words I'm not seeing how you SNAT all other connections to avoid clashing tuples.